SonarQube.Common/ProcessRunner.cs

```diff
@@ -136,6 +130,29 @@ public bool Execute(string exeName, string args, string workingDirectory, int ti
 
         #region Private methods
 
+        private static void SetEnvironmentVariables(ProcessStartInfo psi, IDictionary<string, string> envVariables, ILogger logger)
+        {
+            if (envVariables == null)
+            {
+                return;
+            }
+
+            foreach (KeyValuePair<string, string> envVariable in envVariables)
+            {
+                Debug.Assert(!String.IsNullOrEmpty(envVariable.Key), "Env variable name cannot be null or empty");
+
+                if (psi.EnvironmentVariables.ContainsKey(envVariable.Key))
+                {
+                    logger.LogMessage(Resources.DIAG_Runner_OverwritingEnvVar, envVariable.Key, psi.EnvironmentVariables[envVariable.Key], envVariable.Value);
+                }
+                else
+                {
+                    logger.LogMessage(Resources.DIAG_Runner_SettingEnvVar, envVariable.Key, envVariable.Value);
+                }
+                psi.EnvironmentVariables[envVariable.Key] = envVariable.Value;
```

Comment:
Lgtm. You could create a unit test for the process runner to cover this.

Bogdan

Sent from my Windows Phone
________________________________
From: duncanpMS<mailto:notifications@github.com>
Sent: ‎27/‎06/‎2015 09:07
To: SonarSource/sonar-msbuild-runner<mailto:sonar-msbuild-runner@noreply.github.com>
Subject: Re: [sonar-msbuild-runner] SONARMSBRU-75 Fix dictionary population error (#35)


In SonarQube.Common/ProcessRunner.cs<https://github.com/SonarSource/sonar-msbuild-runner/pull/35#discussion_r33410412>:

> +                return;
> +            }
> +
> +            foreach (KeyValuePair<string, string> envVariable in envVariables)
> +            {
> +                Debug.Assert(!String.IsNullOrEmpty(envVariable.Key), "Env variable name cannot be null or empty");
> +
> +                if (psi.EnvironmentVariables.ContainsKey(envVariable.Key))
> +                {
> +                    logger.LogMessage(Resources.DIAG_Runner_OverwritingEnvVar, envVariable.Key, psi.EnvironmentVariables[envVariable.Key], envVariable.Value);
> +                }
> +                else
> +                {
> +                    logger.LogMessage(Resources.DIAG_Runner_SettingEnvVar, envVariable.Key, envVariable.Value);
> +                }
> +                psi.EnvironmentVariables[envVariable.Key] = envVariable.Value;


This is actual fix - using "[key] = " rather than ".Add(...)"

—
Reply to this email directly or view it on GitHub<https://github.com/SonarSource/sonar-msbuild-runner/pull/35/files#r33410412>.